File: languages/dotnet/src/model/TableAnalysis.cs
using System;
using System.Collections.Generic;

namespace experimento.src.model {
 
    /// <summary>
    /// Interface para analise dos dados
    /// </summary>
    public interface TableAnalysis<T> {      
 
        /// <summary>
        /// Realiza a analise dos dados
        /// </summary>
        /// <param name="userInfoList">Lista de dados a ser analisada</param>
        /// <returns>Resultado da analise</returns>
        T analysis(List<UserInfo> userInfoList);
    }
}

File: languages/python3/src/model/TableAnalysis.py
from src.model.UserInfo import UserInfo

# Interface para analise dos dados
class TableAnalysis:

    # Realiza a analise dos dados
    # @param userInfoList Lista de dados a ser analisada
    # @return Resultado da analise
    def analysis(self, userInfoList):
        raise NotImplementedError()
